subsys/bluetooth/host/classic/sdp.c

```diff
@@ -80,8 +80,19 @@ NET_BUF_POOL_FIXED_DEFINE(sdp_pool, CONFIG_BT_MAX_CONN, BT_L2CAP_BUF_SIZE(SDP_MT
 
 #define SDP_CLIENT_MTU 64
 
+enum __packed sdp_client_state {
```

Comment:
What's the reason you're using `__packed` here? It's not benefitting you at all in the `struct bt_sdp_client` definition (since there'll anyway be padding), so the end result is actually larger code size without any benefit.

Reply:
You *can* use `__packed` when it makes sense (e.g. smaller memory footprint or because the type is required to be of a certain size due to a protocol specification), but I'm not aware of any style or rule to always use it.